lib/salesforce/client.rb

```diff
@@ -0,0 +1,24 @@
+# Eventually, this should be where we have code to talk directly to salesforce
+# instead of using the openstax_salesforce gem.
+# For now - this is just to get the salesforce client to interact with the pardot api
+
+module Salesforce
+  class Client
+    def initialize
+      salesforce_secrets = Rails.application.secrets.salesforce
+
+      salesforce_client = Restforce.new(username: salesforce_secrets[:username],
+                                        password: salesforce_secrets[:password],
+                                        security_token: salesforce_secrets[:security_token],
+                                        client_id: salesforce_secrets[:consumer_key],
+                                        client_secret: salesforce_secrets[:consumer_secret],
+                                        host: salesforce_secrets[:login_domain],
+                                        api_version: '51.0')
+      @client = salesforce_client.authenticate!
```

Comment:
fwiw, I think `authenticate!` returns the [body of a response](https://github.com/restforce/restforce/blob/master/lib/restforce/middleware/authentication.rb#L39), so I wouldn't call it `@client` -- that could be confusing (as it was to me :-).  Maybe `client_data` or something

Reply:
Actually, I figured out this whole file wasn't really needed. I was able to get the `access_token` from OpenStax::Salesforce!